### PR TITLE
Support withServerSideProcessing in SelectWithPredicate

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

Added support for server side processing of predicate filtering in select components. Followed the same pattern we use in TableHOC withServerSideProcessing.

Basically, the idea is that when processing is done by the server, no filtering is applied on the UI side, so we pass down `ownProps.items` directly to the wrapped component. Also, when the predicate value changes, we trigger the `onUpdate` callback so that a call to the server can be sent.

### Potential Breaking Changes

None.

### Acceptance Criteria

-   [x] The proposed changes are covered by unit tests
-   [x] The potential breaking changes are clearly identified
-   [x] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-vapor/src/components/select/tests/MultiSelectWithPredicate.spec.tsx [hunk 1]
```diff
@@ -150,5 +150,28 @@ describe('Select', () => {
                 expect(spy).toHaveBeenCalledWith(unselectListBoxOption(id, items[1].value));
             });
         });
+
+        describe('when predicates are processed on the server side', () => {
```
Comment (gdostie) [inline packages/react-vapor/src/components/select/tests/MultiSelectWithPredicate.spec.tsx]: Forgot to add a test for the onUpdate prop being called when `predicate` changes...

File: packages/react-vapor/src/components/select/tests/MultiSelectWithPredicate.spec.tsx [hunk 2]
```diff
@@ -150,5 +152,44 @@ describe('Select', () => {
...
+                    store
+                )
+                    .dive()
+                    .dive();
+
+                expect(onUpdateSpy).not.toHaveBeenCalled();
+
+                component.setProps({predicate: 'some-other-predicate'});
+
+                expect(onUpdateSpy).not.toHaveBeenCalledTimes(1);
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/select/tests/MultiSelectWithPredicate.spec.tsx]: remove `.not` ?

Comment (gdostie) [reply packages/react-vapor/src/components/select/tests/MultiSelectWithPredicate.spec.tsx]: woops indeed and the test is probably failing then :disappointed:

Comment (gdostie) [reply packages/react-vapor/src/components/select/tests/MultiSelectWithPredicate.spec.tsx]: fixed

File: packages/react-vapor/src/components/select/SelectWithPredicate.tsx [hunk 3]
```diff
@@ -13,44 +17,60 @@ export interface ISelectWithPredicateOwnProps {
...
 
 export const selectWithPredicate = (
-    Component: React.ComponentClass<ISelectProps> | React.StatelessComponent<ISelectProps>
-): React.ComponentClass<ISelectWithPredicateProps> => {
-    const mapStateToProps = (state: IReactVaporState, ownProps: ISelectWithPredicateProps): Partial<ISelectProps> => {
-        const flatSelect = _.findWhere(state.flatSelect, {id: ownProps.id});
-        const predicate = (flatSelect && flatSelect.selectedOptionId) || ownProps.options[0].id;
+    Component: React.ComponentType<ISelectProps>
+): React.ComponentType<ISelectWithPredicateProps> => {
+    const mapStateToProps = (
```
Comment (francoislg) [inline packages/react-vapor/src/components/select/SelectWithPredicate.tsx]: Ça change pas grand chose, mais tu pourrais sortir cette méthode de `selectWithPredicate`. 

Présentement, tu peux techniquement accéder à `Component` et ça serait plus safe d'enlever cette possibilité.

Comment (gdostie) [reply packages/react-vapor/src/components/select/SelectWithPredicate.tsx]: Done

**Pull request merged**